src/plugins/xerces/xerces.cpp

```diff
@@ -0,0 +1,164 @@
+/**
+ * @file this file contains the entry point to the plugin as a gateway between c and c++
+ *
+ * @brief Plugin enables storage to xml files via the Xerces library
+ *
+ * @copyright BSD License (see doc/LICENSE.md or http://www.libelektra.org)
+ */
+
+#include "xerces.hpp"
+#include "deserializer.hpp"
+#include "serializer.hpp"
+#include "util.hpp"
+
+#include <stdio.h>
+
+#include <xercesc/dom/DOM.hpp>
+#include <xercesc/util/OutOfMemoryException.hpp>
+#include <xercesc/util/PlatformUtils.hpp>
+
+using namespace ckdb; // required for kdberrors.h
+
+#include <kdberrors.h>
+#include <kdbhelper.h>
+
+XERCES_CPP_NAMESPACE_USE
+
+int elektraXercesOpen (Plugin * handle, Key * errorKey)
+{
+	XMLPlatformUtils::Initialize ();
+	return 1;
+}
+
+int elektraXercesClose (Plugin * handle, Key * errorKey)
+{
+	XMLPlatformUtils::Terminate ();
```

Comment:
Mmh, surprising how many library developers were unable to program without global variables. Unfortunately, we cannot provide any guarantee from which threads we are called, applications decide that for us. There is not much we can do about it; except to warn applications that they should not call kdbOpen/kdbClose with their own xerces initialization concurrently. We could only use semlock to protect us from concurrent kdbOpen/kdbClose, but that would only protect applications not using xerces themselves. I added this requirement in #1399